Sources/HTTP/HTTPCommon.swift

```diff
@@ -29,20 +29,26 @@ import Foundation
 ///    }
 /// ```
 /// This then needs to be registered with the server using `HTTPServer.start(port:handler:)`
-/// - Parameter req: the incoming HTTP request.
-/// - Parameter res: a writer providing functions to create an HTTP reponse to the request.
-/// - Returns HTTPBodyProcessing: a enum that either discards the request data, or provides a closure to process it.
-public typealias HTTPRequestHandler = (HTTPRequest, HTTPResponseWriter) -> HTTPBodyProcessing
+/// - Parameters:
+///   - req: the incoming HTTP request.
+///   - res: a writer providing functions to create an HTTP reponse to the request.
+///   - queue: optional, if set, dispatch done callbacks back to this queue (if
+///            the handler processing escapes the calling queue)
+/// - Returns: HTTPBodyProcessing: a enum that either discards the request data, or provides a closure to process it.
+public typealias HTTPRequestHandler = (HTTPRequest, HTTPResponseWriter, DispatchQueue?) -> HTTPBodyProcessing
```

Comment:
On Linux you'll need to import Dispatch explicitly to have access to DispatchQueue

Reply:
Aside: from Swift 4.1 that won't be true, thanks to https://github.com/apple/swift-corelibs-foundation/pull/1206 🙂

Reply:
Though importing Foundation is non-sense in the first place, no Foundation is used here ;->